lte/gateway/python/magma/mobilityd/ip_allocator_dhcp.py

```diff
@@ -350,53 +354,39 @@ def alloc_ip_address(self, sid: str, vlan: int) -> IPDesc:
             raise NoAvailableIPError(msg)
 
     def _parse_dhcp_helper_cli_response_to_store(self, dhcp_desc, dhcp_response, mac, vlan):
-        try:
-            # Only look in at the last line of the stdout for the JSON
-            # Previous lines may contain warnings or other unnecessary info.
-            if isinstance(dhcp_response.stdout, str):
-                stdout = dhcp_response.stdout.splitlines()[-1]
-            elif isinstance(dhcp_response.stdout, bytes):
-                stdout = dhcp_response.stdout.decode("utf-8").splitlines()[-1]
-            else:
-                raise TypeError("Invalid type for stdout")
-            dhcp_json = json.loads(stdout)
-        except JSONDecodeError:
-            logging.error(
-                f"Could not decode '{dhcp_response.stdout}' received "
-                f"'{dhcp_response.stderr}' from dhcp_helper_cli called "
-                f"with parameters '{dhcp_response.args}'",
-            )
-            raise NoAvailableIPError(f'Failed to json parse message returned from dhcp_helper_cli.')
-        if dhcp_json:
+        if dhcp_response:
             dhcp_desc = DHCPDescriptor(
                 mac=mac,
-                ip=dhcp_json["ip"],
+                ip=dhcp_response["ip"],
                 vlan=vlan,
                 state_requested=DHCPState.ACK,
                 state=DHCPState.ACK,
-                subnet=str(IPv4Network(dhcp_json["subnet"], strict=False)),
-                server_ip=ip_address(dhcp_json["server_ip"]) if dhcp_json["server_ip"] else None,
-                router_ip=ip_address(dhcp_json["router_ip"]) if dhcp_json["router_ip"] else None,
-                lease_expiration_time=int(dhcp_json["lease_expiration_time"]),
+                subnet=str(IPv4Network(dhcp_response["subnet"], strict=False)),
+                server_ip=ip_address(dhcp_response["server_ip"]) if dhcp_response["server_ip"] else None,
+                router_ip=ip_address(dhcp_response["router_ip"]) if dhcp_response["router_ip"] else None,
+                lease_expiration_time=int(dhcp_response["lease_expiration_time"]),
             )
 
             self._store.dhcp_store[mac.as_redis_key(vlan)] = dhcp_desc
             self._store.dhcp_gw_info.update_ip(dhcp_desc.router_ip, vlan)
         return dhcp_desc
 
     @staticmethod
-    def _get_dhcp_helper_cli_response(call_args):
-        dhcp_response = subprocess.run(
+    def _get_dhcp_helper_cli_response(call_args, save_file: str) -> Dict[str, Any]:
+        ret = subprocess.run(
             *call_args,
             capture_output=True,
         )
-        if dhcp_response.returncode != 0:
+        if ret.returncode != 0:
             logging.error(
-                f"Could not decode '{dhcp_response.stdout}' received"
-                f" '{dhcp_response.stderr}' from {DHCP_HELPER_CLI} called"
-                f" with parameters '{dhcp_response.args}'",
+                f"Could not decode '{ret.stdout}' received"
```

Comment:
I think this error message wording is a little strange now that we no longer parse the stdout.

Reply:
I improved this a bit